content/blog/typescript-type-generics/index.md

```diff
@@ -182,15 +182,15 @@ async function logTheValue(item) {
 }
 ```
 
-If we wanted to type the `logTheValue` function, we'd want to make sure to use a generic for the `loggedValue` was using the same generic type as `item`. We could do so inline:
+If we wanted to add typings for the `logTheValue` function, we'd want to make sure to use a generic for the `loggedValue` <!--Looks like some sentence revisions happened here, not 100% sure what it's supposed to say--> was using the same generic type as `item`. We could do so inline:
 
 ```typescript
-async function logTheValue<ItemT>(item: ItemT): {loggedValue: string, original: ItemT, err: Error | undefined} {
+async function logTheValue<ItemT>(item: ItemT): Promise<{loggedValue: string, original: ItemT, err: Error | undefined}> {
```

Comment:
Ahh, my bad. Let's add a comment explaining why there's a promise here:

```suggestion
// Because this is an `async` function, we want to wrap the returned type value in a Promise
async function logTheValue<ItemT>(item: ItemT): Promise<{loggedValue: string, original: ItemT, err: Error | undefined}> {
```